src/app/dataValidator/helpers/dataLoader/readdirSync.test.js

```diff
@@ -6,14 +6,17 @@ const readScenario = require('./readScenario');
 
 const defaultDataPath = './data';
 
-const expectMethodToBeCalledTimes = (
+const expectMethodToBeCalledTimes = async (
   number,
   spyMethod,
   dataPath = defaultDataPath,
 ) => {
-  readdirSync(dataPath).then(() => {
+  await readdirSync(dataPath);
+  try {
     expect(spyMethod).toHaveBeenCalledTimes(number);
-  });
+  } catch (error) {
```

Comment:
This looks as though you're expecting `expect(spyMethod).toHaveBeenCalledTimes(number);` to sometimes throw an exception...?

If you then call this again in the `catch` block, isn't it going to throw an exception again?

This is basically the same as:

```js
try { }
catch { }
finally {
  expect(spyMethod).toHaveBeenCalledTimes(number);
}
```

... which seems redundant! ^

Reply:
Have now looked at this with Phil and yeah - it's complicated but kind of necessary! He's going to refactor slightly and add a clarifying comment.

Reply:
Seeing this with fresh eyes the try catch isn't needed, good spot!